src/app/machines/components/MachineHeaderForms/MachineActionFormWrapper/CloneForm/CloneFormFields/CloneStorageTable/CloneStorageTable.test.tsx

```diff
@@ -8,60 +6,62 @@ import {
   nodeFilesystem as fsFactory,
   nodePartition as partitionFactory,
 } from "testing/factories";
+import { render, screen, within } from "testing/utils";
 
 describe("CloneStorageTable", () => {
   it("renders empty table if neither loading details nor machine provided", () => {
-    const wrapper = mount(
-      <CloneStorageTable machine={null} selected={false} />
-    );
-    expect(wrapper.find("MainTable").prop("rows")).toStrictEqual([]);
-    expect(wrapper.find("Placeholder").exists()).toBe(false);
+    render(<CloneStorageTable machine={null} selected={false} />);
+    expect(screen.getAllByRole("row")).toHaveLength(1);
+    expect(screen.queryByTestId("placeholder")).not.toBeInTheDocument();
   });
 
   it("renders placeholder content while machine details are loading", () => {
-    const wrapper = mount(
+    render(
       <CloneStorageTable
         loadingMachineDetails
         machine={null}
         selected={false}
       />
     );
-    expect(wrapper.find("Placeholder").exists()).toBe(true);
+    const rows = screen.getAllByRole("row");
+    rows.shift();
+
+    rows.forEach((row) => {
+      const placeholders = within(row).getAllByTestId("placeholder");
+      expect(placeholders[0]).toHaveTextContent("Disk name");
+      expect(placeholders[1]).toHaveTextContent("Model");
+      expect(placeholders[2]).toHaveTextContent("1.0.0");
+      expect(placeholders[3]).toHaveTextContent("Disk type");
+      expect(placeholders[4]).toHaveTextContent("X, X");
+      expect(placeholders[5]).toHaveTextContent("1.23 GB");
+      expect(placeholders[6]).toHaveTextContent("Icon");
+    });
   });
 
   it("renders machine storage details if machine is provided", () => {
-    const machine = machineDetailsFactory({ disks: [diskFactory()] });
-    const wrapper = mount(
-      <CloneStorageTable machine={machine} selected={false} />
-    );
-    expect(wrapper.find("Placeholder").exists()).toBe(false);
-    expect(wrapper.find("MainTable").prop("rows")).not.toStrictEqual([]);
+    const machine = machineDetailsFactory({
+      disks: [diskFactory({ name: "Disk 1" })],
+    });
+    render(<CloneStorageTable machine={machine} selected={false} />);
+    expect(screen.queryByTestId("placeholder")).not.toBeInTheDocument();
+
+    expect(screen.getByText("Disk 1")).toBeInTheDocument();
   });
 
   it("shows a tick for available disks", () => {
     const machine = machineDetailsFactory({
       disks: [diskFactory({ available_size: 1000000000, size: 1000000000 })],
     });
-    const wrapper = mount(
-      <CloneStorageTable machine={machine} selected={false} />
-    );
-
-    expect(
-      wrapper.find("Icon[data-testid='disk-available']").prop("name")
-    ).toBe("tick");
+    render(<CloneStorageTable machine={machine} selected={false} />);
+    expect(screen.getByTestId("disk-available")).toHaveClass("p-icon--tick");
```

Comment:
Would be great if you could add `aria-label="available|not available"` to the Icon component and use it on top of the className assertion.